### ISSUE TITLE
Operator View: Availability - What sleds are up / down?

### ISSUE THREAD

**Issue by smklein** [oxidecomputer/omicron#2035]

**Comment by smklein** [comment]
FWIW, the "sled list" endpoint exists, but it not returning any information about health: https://github.com/oxidecomputer/omicron/blob/75bd90550e63940bbed7ef40ee21e18fa1b628dc/nexus/src/external_api/http_entrypoints.rs#L4972-L5000

Additionally, these sleds remain, even if they are powered off or unplugged

**Comment by davepacheco** [comment]
Presumably part of this includes some kind of state reporting from the Sled itself, storing that in the database, and then using that produce this information in the "list sleds" API?

**Comment by smklein** [comment]
Yeah, definitely. Currently we are storing "what sleds has Nexus ever heard of" in CRDB, through the internal Nexus API:

https://github.com/oxidecomputer/omicron/blob/b8a461754ac302dd6171b76888ad3cbb512a1037/nexus/src/internal_api/http_entrypoints.rs#L66-L86

This... sorta works. It works in cases where the sled info is purely additive, at least? But I don't believe it ever gets updated, and it seems unlikely for a sled to report that it is no longer available (this seems like an inference that needs to be made by Nexus, either by querying MGS, or indirectly through lack-of-a-response).
